app/src/main/java/com/mybus/view/FavoriteNameAlertDialog.java

```diff
@@ -0,0 +1,75 @@
+package com.mybus.view;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+
+import com.mybus.R;
+import com.mybus.listener.OnAddNewFavoriteListener;
+import com.mybus.listener.OnEditOldFavoriteListener;
+
+import butterknife.ButterKnife;
+
+/**
+ * Created by Lucas De Lio on 6/13/2016.
+ */
+public class FavoriteNameAlertDialog extends DialogFragment {
+
+    private OnAddNewFavoriteListener mOnAddNewFavoriteListener;
+    private OnEditOldFavoriteListener mOnEditOldFavoriteListener;
+
+    private String actionType;
+
+    private String mPreviousName;
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        // Get the layout inflater
+        LayoutInflater inflater = getActivity().getLayoutInflater();
+        // Inflate and set the layout for the dialog
+        // Pass null as the parent view because its going in the dialog layout
+        View view = inflater.inflate(R.layout.favorite_name_alert_dialog, null);
+        final EditText mFavofiteNameEditText = ButterKnife.findById(view, R.id.favorite_name_edit_text);
+        if (mPreviousName != null) {
+            mFavofiteNameEditText.setText(mPreviousName);
+        }
+        builder.setView(view)
+                .setPositiveButton(getString(R.string.ok), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        //callback to OnAdd or OnEdit if it's adding or editing the favorite
+                        if (mOnAddNewFavoriteListener != null) {
```

Comment:
Wouldn't be easier to set the type of the dialog such as New or Edit instead of checking the listeners state? Also you can encapsulate both listener into one and only call given the type of the dialog.